src/runtime/test_case_runner.js

```diff
@@ -174,6 +174,22 @@ export default class TestCaseRunner {
     return this.invokeStep(null, hookDefinition, hookParameter)
   }
 
+  async runBeforeHooks(hookDefinitions, hookParameter) {
+    let skipBeforeHooks = false
+    await Promise.each(hookDefinitions, async hookDefinition => {
+      await this.aroundTestStep(async () => {
+        if (skipBeforeHooks) {
+          return { status: Status.SKIPPED }
+        }
+        const hookResult = await this.runHook(hookDefinition, hookParameter)
+        if (hookResult.status === Status.SKIPPED) {
+          skipBeforeHooks = true
+        }
+        return hookResult
+      })
+    })
+  }
+
```

Comment:
Thoughts on the following: update runHooks / runHook to also take in a `isBeforeHook` argument. Then in `runHook` if its a before hook and `isSkippingSteps` returns true, then we return `{ status: Status.SKIPPED }`.

Note if you are curious: `this.skip` in for when the entire scenario is skipped (via dry run or fail fast) and thus no before hooks / after hooks are run